lib/matplotlib/axes/_axes.py

```diff
@@ -4792,6 +4792,8 @@ def _pcolorargs(funcname, *args, **kw):
 
         if len(args) == 3:
             X, Y, C = args
+            X = np.asarray(X)
+            Y = np.asarray(Y)
```

Comment:
should it be `np.asanyarray()`? Don't know if that would make a difference, though. pandas and xray objects, perhaps?